cli/src/diff_util.rs

```diff
@@ -2047,14 +2055,22 @@ pub fn show_diff_stats(
     };
 
     for (stat, ui_path) in iter::zip(stats.entries(), &ui_paths) {
-        let bar_added = (stat.added as f64 * factor).ceil() as usize;
-        let bar_removed = (stat.removed as f64 * factor).ceil() as usize;
+        let bar_length = ((stat.added + stat.removed) as f64 * factor) as usize;
+        // Ensure that any fractional space after scaling is given to whichever
+        // of adds/removes is smaller.
```

Comment:
nit: can you add comment why smaller side is weighed? I think that's in order to print at least one character.

Reply:
Yes, done.